### ISSUE TITLE
`STAT` has an empty DesignAxisRecord

### ISSUE THREAD

**Issue by be5invis** [rsms/inter#308]
The current `STAT` table contain an empty `<DesignAxisRecord>`, however according to the OT spec, implementations may not derive named instances from the rest of the `STAT` table, since they may think this font is static.

**Comment by rsms** [comment]
Do you have an example of where/how I test the effects of this?

**Comment by be5invis** [comment]
@be5invis You can start a windows vm, open your `Inter V.otf` and see whether the previewer shows "previous / next" buttons at top right corner.

**Comment by rsms** [comment]
Thanks @be5invis! That helped. I've revised the patch, now using fonttools instead of a patch file with ttx. Opening the output VF font file in Windows 10 now shows all 18 named instances correctly.

**Comment by be5invis** [comment]
@rsms You could also test on Adobe software (should have VF sliders and 18 named instances) and office softwares (should at least have 18 named instances) to see whether it is correctly recognized.

**Comment by rsms** [comment]
It seemed to work fine in Adobe software (and macOS software like Sketch) before, without the `AxisValueArray` data, so I'm not sure how reliable testing there is. Anyhow, I've got a Windows 10 machine here in my office that I tested on :-)

Here's a build with the changes from d76268cf8d778c709ce2e714a36e17d0d23effa9 which includes an updated STAT table patch:
[Inter.var.ttf.zip](https://github.com/rsms/inter/files/5093519/Inter.var.ttf.zip)

**Comment by be5invis** [comment]
@be5invis The OS could recognize your fonts, but a problem is that: style linking (the **<kbd>b</kbd>** and *<kbd>I</kbd>* buttons) is broken.

Fixing Regular-Bold linking is relatively simple: you could just use AxisValue format 1 for non-regular weights and AxisValue format 3 for Regular with LinkedValue set to 700.

But for Regular-Italics, implementing it is much harder, since your font doesn’t have an `ital` axis. Software like Word may not associate derived italics from uprights and use faux italics when *<kbd>I</kbd>* is used. This is [a well-known problem](https://github.com/arrowtype/recursive/tree/master/docs/03-slnt_ital-axis-rename-tests).

I think for now, the best solution to the italics problem is to use one-axis variable fonts only, and add an `ital` axis **only in `STAT`**, and assign AxisValue's like this:

```xml
<!-- in inter-v-upright.ttf -->
<AxisValue index="9" Format="3">
    <AxisIndex value="1"/>
    <Flags value="2"/>
    <ValueNameID value="279"/> <!-- Regular -->
    <Value value="0.0"/>
    <LinkedValue value="1.0"/>
</AxisValue>

<!-- in inter-v-italic.ttf -->
<AxisValue index="9" Format="1">
    <AxisIndex value="1"/>
    <Flags value="2"/>
    <ValueNameID value="280"/>  <!-- Italics -->
    <Value value="1.0"/>
</AxisValue>
```

**Comment by be5invis** [comment]
I think for the italics problem the currently best solution is to use single-axis fonts for desktop, and follow the STAT table in Source Sans Pro’s variable fonts, which correctly implemented style linking.

**Comment by be5invis** [comment]
And for two axes, well, you can use ... Oblique instead of Italics if you continue using `slnt` for axis tag, or switch `slnt` to `ital` and re-arrange the design space, since `ital` varies from `0` to `1`.

**Comment by rsms** [comment]
With ddb81e26463c533443128ab7a98aba708e58bd1f these are the results in practice:

<img width="1469" alt="Screen Shot 2020-08-19 at 15 51 41" src="https://user-images.githubusercontent.com/47046/90697773-fcc1f300-e233-11ea-98a8-bc007b4ede3d.png">

This is what TTX reports for the STAT table:

```xml
<STAT>
  <Version value="0x00010002"/>
  <DesignAxisRecordSize value="8"/>
  <!-- DesignAxisCount=2 -->
  <DesignAxisRecord>
    <Axis index="0">
      <AxisTag value="wght"/>
      <AxisNameID value="271"/>  <!-- Weight -->
      <AxisOrdering value="0"/>
    </Axis>
    <Axis index="1">
      <AxisTag value="slnt"/>
      <AxisNameID value="272"/>  <!-- Slant -->
      <AxisOrdering value="1"/>
    </Axis>
  </DesignAxisRecord>
  <!-- AxisValueCount=11 -->
  <AxisValueArray>
    <AxisValue index="0" Format="4">
      <!-- AxisCount=2 -->
      <Flags value="2"/>
      <ValueNameID value="279"/>  <!-- Regular -->
      <AxisValueRecord index="0">
        <AxisIndex value="0"/>
        <Value value="400.0"/>
      </AxisValueRecord>
      <AxisValueRecord index="1">
        <AxisIndex value="1"/>
        <Value value="0.0"/>
      </AxisValueRecord>
    </AxisValue>
    <AxisValue index="1" Format="4">
      <!-- AxisCount=2 -->
      <Flags value="0"/>
      <ValueNameID value="280"/>  <!-- Italic -->
      <AxisValueRecord index="0">
        <AxisIndex value="0"/>
        <Value value="400.0"/>
      </AxisValueRecord>
      <AxisValueRecord index="1">
        <AxisIndex value="1"/>
        <Value value="-10.0"/>
      </AxisValueRecord>
    </AxisValue>
    <AxisValue index="2" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="273"/>  <!-- Thin -->
      <Value value="100.0"/>
    </AxisValue>
    <AxisValue index="3" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="275"/>  <!-- Extra Light -->
      <Value value="200.0"/>
    </AxisValue>
    <AxisValue index="4" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="277"/>  <!-- Light -->
      <Value value="300.0"/>
    </AxisValue>
    <AxisValue index="5" Format="1">
      <AxisIndex value="0"/>
      <Flags value="2"/>
      <ValueNameID value="279"/>  <!-- Regular -->
      <Value value="400.0"/>
    </AxisValue>
    <AxisValue index="6" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="281"/>  <!-- Medium -->
      <Value value="500.0"/>
    </AxisValue>
    <AxisValue index="7" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="283"/>  <!-- Semi Bold -->
      <Value value="600.0"/>
    </AxisValue>
    <AxisValue index="8" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="285"/>  <!-- Bold -->
      <Value value="700.0"/>
    </AxisValue>
    <AxisValue index="9" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="287"/>  <!-- Extra Bold -->
      <Value value="800.0"/>
    </AxisValue>
    <AxisValue index="10" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="289"/>  <!-- Black -->
      <Value value="900.0"/>
    </AxisValue>
  </AxisValueArray>
  <ElidedFallbackNameID value="2"/>  <!-- Regular -->
</STAT>
```

**Comment by be5invis** [comment]
@rsms The settings image is actually wrong: items are duplicating.
The following `STAT` will create a proper list.

```xml
<STAT>
  <Version value="0x00010002"/>
  <DesignAxisRecordSize value="8"/>
  <!-- DesignAxisCount=2 -->
  <DesignAxisRecord>
    <Axis index="0">
      <AxisTag value="wght"/>
      <AxisNameID value="271"/>  <!-- Weight -->
      <AxisOrdering value="0"/>
    </Axis>
    <Axis index="1">
      <AxisTag value="slnt"/>
      <AxisNameID value="272"/>  <!-- Slant -->
      <AxisOrdering value="1"/>
    </Axis>
  </DesignAxisRecord>
  <!-- AxisValueCount=11 -->
  <AxisValueArray>
    <AxisValue index="0" Format="3">
     <AxisIndex value="1"/>
      <Flags value="2"/>
      <ValueNameID value="279"/>  <!-- Regular -->
      <Value value="0.0"/>
      <LinkedValue value="-10.0"/>
    </AxisValue>
    <AxisValue index="1" Format="1">
      <AxisIndex value="1"/>
      <Flags value="0"/>
      <ValueNameID value="280"/>  <!-- Italic -->
      <Value value="-10.0"/>
    </AxisValue>
    <AxisValue index="2" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="273"/>  <!-- Thin -->
      <Value value="100.0"/>
    </AxisValue>
    <AxisValue index="3" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="275"/>  <!-- Extra Light -->
      <Value value="200.0"/>
    </AxisValue>
    <AxisValue index="4" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="277"/>  <!-- Light -->
      <Value value="300.0"/>
    </AxisValue>
    <AxisValue index="5" Format="1">
      <AxisIndex value="0"/>
      <Flags value="2"/>
      <ValueNameID value="279"/>  <!-- Regular -->
      <Value value="400.0"/>
    </AxisValue>
    <AxisValue index="6" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="281"/>  <!-- Medium -->
      <Value value="500.0"/>
    </AxisValue>
    <AxisValue index="7" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="283"/>  <!-- Semi Bold -->
      <Value value="600.0"/>
    </AxisValue>
    <AxisValue index="8" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="285"/>  <!-- Bold -->
      <Value value="700.0"/>
    </AxisValue>
    <AxisValue index="9" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="287"/>  <!-- Extra Bold -->
      <Value value="800.0"/>
    </AxisValue>
    <AxisValue index="10" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="289"/>  <!-- Black -->
      <Value value="900.0"/>
    </AxisValue>
  </AxisValueArray>
  <ElidedFallbackNameID value="2"/>  <!-- Regular -->
</STAT>
```

![image](https://user-images.githubusercontent.com/240091/90701949-df465680-e23e-11ea-9efb-16fd95c57ff7.png)

**Comment by be5invis** [comment]
@rsms 
The weight 400 could also use format 3 value to get linked with Bold.

```xml
<STAT>
  <Version value="0x00010002"/>
  <DesignAxisRecordSize value="8"/>
  <!-- DesignAxisCount=2 -->
  <DesignAxisRecord>
    <Axis index="0">
      <AxisTag value="wght"/>
      <AxisNameID value="271"/>  <!-- Weight -->
      <AxisOrdering value="0"/>
    </Axis>
    <Axis index="1">
      <AxisTag value="slnt"/>
      <AxisNameID value="272"/>  <!-- Slant -->
      <AxisOrdering value="1"/>
    </Axis>
  </DesignAxisRecord>
  <!-- AxisValueCount=11 -->
  <AxisValueArray>
    <AxisValue index="0" Format="3">
     <AxisIndex value="1"/>
      <Flags value="2"/>
      <ValueNameID value="279"/>  <!-- Regular -->
      <Value value="0.0"/>
      <LinkedValue value="-10.0"/>
    </AxisValue>
    <AxisValue index="1" Format="1">
      <AxisIndex value="1"/>
      <Flags value="0"/>
      <ValueNameID value="280"/>  <!-- Italic -->
      <Value value="-10.0"/>
    </AxisValue>
    <AxisValue index="2" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="273"/>  <!-- Thin -->
      <Value value="100.0"/>
    </AxisValue>
    <AxisValue index="3" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="275"/>  <!-- Extra Light -->
      <Value value="200.0"/>
    </AxisValue>
    <AxisValue index="4" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="277"/>  <!-- Light -->
      <Value value="300.0"/>
    </AxisValue>
    <AxisValue index="5" Format="3">
      <AxisIndex value="0"/>
      <Flags value="2"/>
      <ValueNameID value="279"/>  <!-- Regular -->
      <Value value="400.0"/>
      <LinkedValue value="700.0"/>
    </AxisValue>
    <AxisValue index="6" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="281"/>  <!-- Medium -->
      <Value value="500.0"/>
    </AxisValue>
    <AxisValue index="7" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="283"/>  <!-- Semi Bold -->
      <Value value="600.0"/>
    </AxisValue>
    <AxisValue index="8" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="285"/>  <!-- Bold -->
      <Value value="700.0"/>
    </AxisValue>
    <AxisValue index="9" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="287"/>  <!-- Extra Bold -->
      <Value value="800.0"/>
    </AxisValue>
    <AxisValue index="10" Format="1">
      <AxisIndex value="0"/>
      <Flags value="0"/>
      <ValueNameID value="289"/>  <!-- Black -->
      <Value value="900.0"/>
    </AxisValue>
  </AxisValueArray>
  <ElidedFallbackNameID value="2"/>  <!-- Regular -->
</STAT>
```

**Comment by rsms** [comment]
Updated the patching code to create a table like that @be5invis with linking (nice!) and it seems to work well.

Windows seems to be the only app affected by these changes, by the way. 

<img width="278" alt="Screen Shot 2020-08-19 at 17 46 42" src="https://user-images.githubusercontent.com/47046/90704908-91355100-e246-11ea-9bc3-fe229a0a3d94.png">

**Comment by rsms** [comment]
Here's a build of master (f7924a233e82f71872c31eaa2093401a74f35460):
[Inter-3.15-text-f7924a233e.zip](https://github.com/rsms/inter/files/5100101/Inter-3.15-text-f7924a233e.zip)

**Comment by be5invis** [comment]
@rsms This one worked well. Closing :)

**Comment by rsms** [comment]
Thank you @be5invis for all your help on this!

**Comment by rsms** [comment]
This has now been released as part of [version 3.15](https://github.com/rsms/inter/releases/tag/v3.15) 🎉
Try it out here: https://rsms.me/inter/lab/

**Comment by Lorp** [comment]
Looks good :)
